shared/events.js

```diff
@@ -402,9 +364,21 @@ export class PostMessage extends HashableAction {
   }
 }
 
-export class AuthorizeSender extends HashableAction {
+export class HashableMailboxAuthorizeSender extends HashableAction {
   static authorization = CanModifyAuths
-  static fields = AuthorizeSender.Fields([
+  static fields = HashableMailboxAuthorizeSender.Fields([
     ['sender', 'string']
   ])
 }
+
+// Function for converting from
+export function ConvertToBackendEntry (entry, hash) {
```

Comment:
Could you add a TODO above this to rewrite this function in a generic/automatic way so that a switch statement isn't needed and new classes are automatically converted based on the naming convention?